src/cpp/fastdds/publisher/PublisherImpl.cpp

```diff
@@ -532,13 +532,83 @@ const Publisher* PublisherImpl::get_publisher() const
     return user_publisher_;
 }
 
-/* TODO
-   bool PublisherImpl::delete_contained_entities()
-   {
-    logError(PUBLISHER, "Operation not implemented");
-    return false;
-   }
- */
+ReturnCode_t PublisherImpl::delete_contained_entities()
+{
+    // Let's be optimistic
+    ReturnCode_t result = ReturnCode_t::RETCODE_OK;
+
+    bool can_be_deleted = true;
+
+    std::vector<DataWriter*> writer_vector;
+    get_datawriters(writer_vector);
+
+    std::lock_guard<std::mutex> lock(mtx_writers_);
+    for (DataWriter* writer: writer_vector)
+    {
+        can_be_deleted = can_be_deleted && (writer->impl_->check_delete_preconditions() == ReturnCode_t::RETCODE_OK);
+        if (!can_be_deleted)
+        {
+            return ReturnCode_t::RETCODE_PRECONDITION_NOT_MET;
+        }
+    }
+
+    // We traverse the map trying to delete all writers;
+    auto writer_iterator = writers_.begin();
+    while (writer_iterator != writers_.end())
+    {
+        //First extract the writer from the maps to free the mutex
+        auto it = writer_iterator->second.begin();
+        DataWriterImpl* writer_impl = *it;
+        ReturnCode_t ret_code = writer_impl->check_delete_preconditions();
+        if (!ret_code)
+        {
+            return ret_code;
```

Comment:
This should return a `RETCODE_ERROR` as in the subscriber. We already checked the writers can be deleted, and if now we cannot, something went wrong.